File: java/web_tutorial_mastercopy/reddit_snippet.java
import java.util.ArrayList;
import bridges.connect.Bridges;
import bridges.connect.DataSource;
import bridges.data_src_dependent.Reddit;


// This program fragment illustrates how to access the reddit data
public class reddit_snippet {
	public static void main(String[] args) throws Exception {

		// create Bridges object
#if TESTING
		// command line args provide credentials and server type
		Bridges bridges = new Bridges(Integer.parseInt(args[0]), args[1], args[2]);
		bridges.setServer(args[3]);
#else
		Bridges bridges = new Bridges (YOUR_ASSSIGNMENT_NUMBER, "YOUR_USER_ID", "YOUR_API_KEY");
#endif
    
		DataSource ds = bridges.getDataSource();

		ArrayList<String> subreddits = ds.getAvailableSubreddits();

		// get some data from the  "Ask Science" subreddit
		ArrayList<Reddit> reddit_posts = ds.getRedditData("askscience");
		int count = 0;
		for (Reddit r: reddit_posts) {
		    ++count;
		    if (count > 10)
			break;
		    System.out.println("ID: "+r.getID());
		    System.out.println("Title: "+r.getTitle());
		    System.out.println("Author: "+r.getAuthor());
		    System.out.println("Score: "+r.getScore());
		    System.out.println("VoteRatio: "+r.getVoteRatio());
		    System.out.println("CommentCount: "+r.getCommentCount());
		    System.out.println("Subreddit: "+r.getSubreddit());
		    System.out.println("PostTime: "+r.getPostTime());
		    System.out.println("URL: "+r.getURL());
		    System.out.println("Text: "+r.getText());
		    System.out.println("");
		}
		
	}
}

File: python/web_tutorial_mastercopy/reddit_snippet.py
from bridges.bridges import *
from bridges.color import *
from bridges.data_src_dependent import data_source
import sys
import os

def main():

#if TESTING
    # command line args provide credentials and server to test on
    args = sys.argv[1:]
    bridges = Bridges(int(args[0]), args[1], args[2])
    if len(args) > 3:
        bridges.connector.set_server(args[3])
#else
    bridges = Bridges(YOUR_ASSSIGNMENT_NUMBER, "YOUR_USER_ID", "YOUR_API_KEY")
#endif


    reddit_obj = data_source.reddit_data("askscience")
    
    for i in range(0, 3):
        print(f"Title: {reddit_obj[i].title}")
        print(f"Author: {reddit_obj[i].author}")
        print(f"Score: {reddit_obj[i].score}")
        print(f"Vote Ratio: {reddit_obj[i].vote_ratio}")
        print(f"Comment Count: {reddit_obj[i].comment_count}")
        print(f"Subreddit: {reddit_obj[i].subreddit}")
        print(f"PostTime: {reddit_obj[i].post_time}")
        print(f"URL: {reddit_obj[i].url}")
        print(f"Text: {reddit_obj[i].text}")
        print("")


if __name__ == "__main__":
    main()
